### DISCUSSION TITLE
how to automatically register new airflow users when they first login

### DISCUSSION THREAD

**Discussion by kangtao** [apache/airflow#14411] [Q&A]
Hi,

I am reading through airflow OAuth Authentication docs from https://airflow.apache.org/docs/apache-airflow/1.10.1/security.html#oauth-authentication, and Flask-AppBuilder OAuth Authentication docs https://flask-appbuilder.readthedocs.io/en/latest/security.html. From what I understand, after the user log in,  the framework retrieves information about the user, username and email.  This info will be checked with the internal user (user record on User Model), first by username next by email. Does it mean that in order to sign user in, we need to pre-create the user in airflow DB before first sign in? Is it possible for airflow to register the new user from the user's first login without pre-create users?

Thanks!

**Comment by turbaszek** [comment]
> Does it mean that in order to sign user in, we need to pre-create the user in airflow DB before first sign in?

Yes, you need at least one admin user. After that you can add users via WebUI or using `airflow users` command

**Reply by kangtao** [reply]
Thanks for your reply. Do you know how to register new users when they first login? Do I need to override some methods in AirflowSecurityManager?
